src/Umbraco.Web.Common/Runtime/AspNetCoreComposer.cs

```diff
@@ -39,41 +41,19 @@ public new void Compose(Composition composition)
             composition.RegisterUnique<IApplicationShutdownRegistry, AspNetCoreApplicationShutdownRegistry>();
 
             // The umbraco request lifetime
-            composition.RegisterUnique<UmbracoRequestLifetime>();
-            composition.RegisterUnique<IUmbracoRequestLifetimeManager>(factory => factory.GetInstance<UmbracoRequestLifetime>());
-            composition.RegisterUnique<IUmbracoRequestLifetime>(factory => factory.GetInstance<UmbracoRequestLifetime>());
+            composition.RegisterMultipleUnique<IUmbracoRequestLifetime, IUmbracoRequestLifetimeManager, UmbracoRequestLifetime>();
 
-            composition.RegisterUnique<IPasswordHasher, AspNetCorePasswordHasher>();
-
-
-            // register the http context and umbraco context accessors
-            // we *should* use the HttpContextUmbracoContextAccessor, however there are cases when
-            // we have no http context, eg when booting Umbraco or in background threads, so instead
-            // let's use an hybrid accessor that can fall back to a ThreadStatic context.
-            composition.RegisterUnique<IUmbracoContextAccessor, HybridUmbracoContextAccessor>();
 
-            // register the umbraco context factory
-            // composition.RegisterUnique<IUmbracoContextFactory, UmbracoContextFactory>();
-            composition.RegisterUnique<IPublishedUrlProvider, UrlProvider>();
-
-            composition.RegisterUnique<HtmlLocalLinkParser>();
-            composition.RegisterUnique<HtmlImageSourceParser>();
-            composition.RegisterUnique<HtmlUrlParser>();
-            composition.RegisterUnique<RichTextEditorPastedImages>();
+            //Password hasher
+            composition.RegisterUnique<IPasswordHasher, AspNetCorePasswordHasher>();
 
-            composition.UrlProviders()
-                .Append<AliasUrlProvider>()
-                .Append<DefaultUrlProvider>();
 
-            composition.MediaUrlProviders()
-                .Append<DefaultMediaUrlProvider>();
+            composition.RegisterUnique<ICookieManager, AspNetCoreCookieManager>();
 
-            composition.RegisterUnique<ISiteDomainHelper, SiteDomainHelper>();
+            composition.RegisterMultipleUnique<ISessionIdResolver, ISessionManager, AspNetCoreSessionManager>();
 
-            // register properties fallback
-            composition.RegisterUnique<IPublishedValueFallback, PublishedValueFallback>();
+            composition.RegisterUnique<ISessionIdResolver, AspNetCoreSessionManager>();
```

Comment:
Is this a duplicate since above there is already `composition.RegisterMultipleUnique<ISessionIdResolver, ISessionManager, AspNetCoreSessionManager>();` ?